services/httpd/handler.go

```diff
@@ -1062,6 +1062,10 @@ func (h *Handler) servePromRead(w http.ResponseWriter, r *http.Request, user met
 		h.httpError(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	if rs == nil {
+		h.httpError(w, "no result set found", http.StatusNotFound)
```

Comment:
I'll need more information from @zeebo but I suspect that this condition is not really the client's fault. Probably there just isn't any data. I wonder if the status code should be a `204`.